sql/src/test/java/io/crate/jobs/PageDownstreamContextTest.java

```diff
@@ -186,4 +187,49 @@ public void testSetBucketOnAKilledCtxReleasesListener() throws Exception {
         // Must not timeout
         listenerReleased.get(1, TimeUnit.SECONDS);
     }
+
+    @Test
+    public void testNonSequentialBucketIds() throws Exception {
```

Comment:
this test will succeed without any changes applied

Reply:
Yes, we can always add more tests, no? It's an additional precaution, please revert back to the first commit supplied in the description, then it will fail. And it was merged only yesterday :)

Reply:
so this commit then an optimized version of https://github.com/crate/crate/commit/dd4ba3a436f93e68ec6e8f267e9a96c305bdc017 ? because the commit message describes this a bit different ;)